packages/vm/lib/evm/opFns.ts

```diff
@@ -566,6 +566,32 @@ export const handlers: { [k: string]: OpHandler } = {
     runState.stack.push(new BN(runState.eei.getGasLeft()))
   },
   JUMPDEST: function (runState: RunState) {},
+  BEGINSUB: function (runState: RunState) {
+    trap(ERROR.INVALID_BEGINSUB + ' at ' + describeLocation(runState))
```

Comment:
Note: the spec says that walking into a BEGINSUB should OOG. 
> Attempted execution of a BEGINSUB causes an abort: terminate execution with an OOG (Out Of Gas) exception

However, both geth and besu have implemented a custom error for this: `invalid subroutine entry`
+ [geth][1]
+ [besu][2]

OpenEthereum's pending PR for the BEGINSUB behavior update uses OOG
+ [OE][3]

[1]: https://github.com/ethereum/go-ethereum/pull/20619/files#diff-8da884c4028661214a82aa91a60e525eR28
[2]: https://github.com/hyperledger/besu/pull/941/files#diff-1c430c8ef6d9f112d7e8703dc9ea0ce5R27
[3]: https://github.com/openethereum/openethereum/pull/11731/files#diff-f24406c9371a7a621c14519c77ce4bddR556

Reply:
@holiman @matkt @adria0 What's the consensus here?

Reply:
mmm...I think that OE should create a new type of error that throws an OOG. Let's me check this.

Reply:
`OOG` is correct. As in: for consensus purposes, it behaves in all respects _exactly_ as an OOG. The actual text output of the error is not part of consensus.

Reply:
Ah! Ok, thanks.